rhel/vex/parser.go

```diff
@@ -508,12 +601,91 @@ func createPackageKey(repo, name, fixedIn string) string {
 	return repo + ":" + name + "-" + fixedIn
 }
 
-func epochVersion(purl *packageurl.PackageURL) string {
-	epoch := "0"
-	if e, ok := purl.Qualifiers.Map()["epoch"]; ok {
-		epoch = e
+// ExtractFixedInVersion deals with 2 pURL types, TypeRPM and TypeOCI
+//   - TypeOCI: return the tag qualifier.
+//   - TypeRPM: check for an epoch qualifier and prepend it to the purl.Version.
+//     If no epoch qualifier, default to 0.
+func extractFixedInVersion(purl packageurl.PackageURL) (string, error) {
+	switch purl.Type {
+	case packageurl.TypeOCI:
+		t, ok := purl.Qualifiers.Map()["tag"]
+		if !ok {
+			return "", fmt.Errorf("could not find tag qualifier for OCI purl type %s", purl.String())
+		}
+		return t, nil
+	case packageurl.TypeRPM:
+		epoch := "0"
+		if e, ok := purl.Qualifiers.Map()["epoch"]; ok {
+			epoch = e
+		}
+		return epoch + ":" + purl.Version, nil
+	default:
+		return "", fmt.Errorf("unexpected purl type %s", purl.Type)
+	}
+}
+
+// ExtractPackageName deals with 2 pURL types, TypeRPM and TypeOCI
+//   - TypeODI: check if there is Namespace and Name i.e. rhel7/rhel-atomic
+//     and return that, if not, check for a repository_url qualifier. If the
+//     repository_url exists then use the namespace/name part, if not, use
+//     the purl.Name.
+//   - TypeRPM: Just return the purl.Name.
+func extractPackageName(purl packageurl.PackageURL) (string, error) {
+	switch purl.Type {
+	case packageurl.TypeOCI:
+		if purl.Namespace != "" {
+			return purl.Namespace + "/" + purl.Name, nil
+		}
+		// Try finding an image name from the tag qualifier
+		ru, ok := purl.Qualifiers.Map()["repository_url"]
+		if !ok {
+			return purl.Name, nil
+		}
+		r := strings.SplitN(ru, "/", 2)
```

Comment:
nit: this is fine, but I tend to prefer `strings.Cut` instead of `strings.SplitN(<whatever>, <whatever>, 2)`

Reply:
yeah, better